pkg/commands/git_commands/commit.go

```diff
@@ -22,7 +22,13 @@ func NewCommitCommands(gitCommon *GitCommon) *CommitCommands {
 
 // ResetAuthor resets the author of the topmost commit
 func (self *CommitCommands) ResetAuthor() error {
+	message, err := self.GetCommitMessage("HEAD")
```

Comment:
It seems a bit wasteful to ask git for the commit message here when we have it in our model in memory already. Maybe we should pass in either the subject or a `skipHook` bool into this function (and the other ones below).

Reply:
I'm not entirely sure we do, there's [this](https://github.com/jesseduffield/lazygit/blob/b2ff09ec0cb8e10dad405c1d586aa09c8f073c2b/pkg/commands/git_commands/commit.go#L43) in the same file so I figured we don't have it ready. I can't find a place where we actually store the current message.

Reply:
> there's [this](https://github.com/jesseduffield/lazygit/blob/b2ff09ec0cb8e10dad405c1d586aa09c8f073c2b/pkg/commands/git_commands/commit.go#L43) in the same file so I figured we don't have it ready.

That code needs to get the message from git because it needs the full message, including the body (what we call "description" in lazygit). You don't need that here, the subject is enough in this case, and you can just take that from the models.Commit object.

Reply:
Could you please point me in the direction of how to get `models` in this context?

We have `Model()` in `ControllerCommon`, but `CommitCommands` has `GitCommon` and I can't find anything resembling `Model` there, not even through code-completion.

Reply:
The immediate caller (`RebaseCommands.ResetCommitAuthor`) has the commit and could pass it in; either as the subject string, or as a `skipHook` bool (by doing the HasPrefix at the call site), as I said above. That's the simplest way.

However, I'd suggest to do it one level above in the call chain (`LocalCommitsController.resetAuthor`) and pass in the information from there; probably rather as a skipHooks flag than passing the subject. My thinking is that here we are in the `git_commands` package, and this is only responsible for carrying out git commands, but not have any logic otherwise; decisions like that should be made by clients and passed in as flags.

You probably want to have a `shouldSkipCommitHooks(models.Commit *commit)` helper function in `LocalCommitsController` to avoid the code repetition.